library-best-practices/JLBP-5.md

```diff
@@ -36,8 +36,14 @@ Example 2: There are multiple artifacts that provide classes under
     might be used anywhere that is not a servlet. Define these status codes
     yourself or choose a different library to provide them.
 
-Problems caused by overlapping classes can be extremely difficult to resolve.
-If at all possible, eliminate all but one of the
-overlapping dependencies. Otherwise pay very close attention to which version of
-each overlapping class is chosen. Make sure the project does not depend on any
-behavior or API of the class that is not selected.
+In Java 9 and later overlapping classes become compile-time and runtime errors when
+named modules are used. It is critical, especially in Java 9 and later,
+to remove all but one of the overlapping classes from the classpath.
+
+If possible, upgrade all dependencies on the overlapping classes
```

Comment:
The wording "upgrade all dependencies on the overlapping classes" sounds odd and requires a re-interpretation in the middle of the sentence. I would suggest: "upgrade all dependency declarations on the artifacts containing the overlapping classes".

Reply:
rewritten